optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -186,20 +186,109 @@
     // Group by
     // ************************************************************************
 
-    <GroupKey_, ResultContainer_, Result_> UniConstraintStream<Result_> groupBy(
+    /**
+     * Runs all items in the stream through a given @{@link UniConstraintCollector} and converts them into a new
+     * {@link UniConstraintStream} which only has one item in it, the result of applying
+     * {@link UniConstraintCollector#finisher()}.
+     *
+     * @param collector never null, the collector to perform the grouping operation with
+     * @param <ResultContainer_> the mutable accumulation type (often hidden as an implementation detail)
+     * @param <Result_> the type of a match fact in the destination {@link UniConstraintStream}
+     * @return never null.
+     */
+    <ResultContainer_, Result_> UniConstraintStream<Result_> groupBy(
             UniConstraintCollector<A, ResultContainer_, Result_> collector);
 
-    <GroupKey_> UniConstraintStream<GroupKey_> groupBy(
-            Function<A, GroupKey_> groupKeyMapping);
-
+    /**
+     * Convert the {@link UniConstraintStream} to a different {@link UniConstraintStream}, consisting of the set of
+     * values resulting from applying the group key mapping function on all items in the original stream. None of the
+     * items in the new stream will {@link Objects#equals(Object, Object)} any other.
```

Comment:
Would you mind doing me a favor for maintenance and start new sensenance on a new line, both in javadoc and asciidoc?
In both cases, those newlines get replaced by a space in the output, so it becomes a paragraph again.
But for editing, at most one sentence per line, makes it really easy to change that line without having to adjust wrapping afterwards for the rest of that paragraph.

Reply:
Also one sentance per line creates "ventilzated proze", see
https://asciidoctor.org/docs/asciidoc-recommended-practices/
https://vanemden.wordpress.com/2009/01/01/ventilated-prose/